src/shared/cache/GenericAssayMolecularDataCache.ts

```diff
@@ -60,7 +60,7 @@ function handleValueThreshold(
 ): GenericAssayDataEnhanced {
     if (!datum.thresholdType) {
         // this is to prevent repeated extraction of '>' and '<' symbols
-        const matches = /([><]?)(.+)/.exec(datum.value);
+        const matches = /([><]?)(-?\\d+(\\.\\d+)?)/.exec(datum.value);
```

Comment:
Previous code will match any string regardless it's numeric or not. That was correct at that moment, but we have free text data now. So we need to change this to match numeric data only.